recipes/drogon/all/conanfile.py

```diff
@@ -0,0 +1,162 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import functools
+
+required_conan_version = ">=1.43.0"
+
+class DrogonConan(ConanFile):
+    name = "drogon"
+    description = "A C++14/17/20 based HTTP web application framework running on Linux/macOS/Unix/Windows"
+    topics = ("http-server", "non-blocking-io", "http-framework", "asynchronous-programming")
+    license = "MIT"
+    homepage = "https://github.com/drogonframework/drogon"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package_multi"
+    options = {
+        "shared": [False, True],
+        "fPIC": [True, False],
+        "with_ctl": [True, False],
+        "with_orm": [True, False],
+        "with_profile": [True, False],
+        "with_brotli": [True, False],
+        "with_postgres": [True, False],
+        "with_postgres_batch": [True, False],
+        "with_mysql": [True, False],
+        "with_sqlite": [True, False],
+        "with_redis": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ctl": False,
+        "with_orm": True,
+        "with_profile": False,
+        "with_brotli": False,
+        "with_postgres": False,
+        "with_postgres_batch": False,
+        "with_mysql": False,
+        "with_sqlite": False,
+        "with_redis": False,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if not self.options.with_orm:
+            del self.options.with_postgres
+            del self.options.with_postgres_batch
+            del self.options.with_mysql
+            del self.options.with_sqlite
+            del self.options.with_redis
+        elif not self.options.with_postgres:
+            del self.options.with_postgres_batch
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            self.options["trantor"].shared = True
```

Comment:
```suggestion
            self.options["trantor"].shared = True
```
Any error which enforces trantor as shared?

Reply:
This is the way of upstream CMakeLists.txt.
https://github.com/drogonframework/drogon/blob/v1.7.5/CMakeLists.txt#L54-L55

Reply:
Good catch!